swap/src/bob.rs

```diff
@@ -142,13 +105,19 @@ pub async fn swap(
     };
 
     swarm.send_message2(alice.clone(), state2.next_message());
+    // NOTE: We must poll the swarm after `send_messageX` to actually trigger
```

Comment:
Did you chat about that with @thomaseizinger? :)

Reply:
Nope, I never got to the bottom of this.

Reply:
You should definitely be polling the swarm outside of this in a loop somewhere. In a production application, the swarm can emit all kinds of events all the time, not just the message you are waiting for.

There are several ways of how you can do this, one of them being the way nectar does things and connect the swaps with the main loop through a channel.

Reply:
To receive a message you will probably want to have a channel in the opposite direction. The sending end could be sent as part of the first message.

Reply:
This is effectively creating an ad-hoc actor system, so you might also want to consider using one.